File: backend/product_recommendations/controllers/recommendation_controller.py
# Epic Title: Implement Product Recommendations Based on User Preferences

from flask import Blueprint, request, jsonify
from backend.product_recommendations.services.recommendation_service import generate_recommendations

recommendation_bp = Blueprint('recommendation', __name__)

@recommendation_bp.route('/generate/<int:user_id>', methods=['POST'])
def generate(user_id):
    try:
        recommendations = generate_recommendations(user_id)
        return jsonify(recommendations), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

File: backend/product_recommendations/controllers/recommendationController.js
// Epic Title: Implement Product Recommendations Based on User Preferences

const express = require('express');
const router = express.Router();
const { getRecommendations } = require('../services/recommendationService');

// GET personalized recommendations
router.get('/:userId', async (req, res) => {
  const { userId } = req.params;
  try {
    const recommendations = await getRecommendations(userId);
    res.status(200).json(recommendations);
  } catch (error) {
    res.status(500).json({ error: 'Failed to fetch recommendations' });
  }
});

module.exports = router;